tests/handlers/records-query.spec.ts

```diff
@@ -202,6 +202,133 @@ export function testRecordsQueryHandler(): void {
+        expect(notOwnerPublishedPostReply.status.code).to.equal(200);
+        expect(notOwnerPublishedPostReply.entries?.length).to.equal(1);
+        expect(notOwnerPublishedPostReply.entries![0].recordId).to.equal(publishedWrite.message.recordId);
+
+        // anonymous query for published records
+        const anonymousPostQuery = await RecordsQuery.create({ filter: { schema: 'post', published: true } });
+        let anonymousPublishedPostReply = await dwn.processMessage(alice.did, anonymousPostQuery.message);
+        expect(anonymousPublishedPostReply.status.code).to.equal(200);
+        expect(anonymousPublishedPostReply.entries?.length).to.equal(1);
+        expect(anonymousPublishedPostReply.entries![0].recordId).to.equal(publishedWrite.message.recordId);
+
+        // publish the unpublished record
+        const publishedDraftWrite = await RecordsWrite.createFrom({
+          recordsWriteMessage : draftWrite.message,
+          published           : true,
+          authorizationSigner : Jws.createSigner(alice)
+        });
+        const publishedDraftReply = await dwn.processMessage(alice.did, publishedDraftWrite.message);
+        expect(publishedDraftReply.status.code).to.equal(202);
+
+        // issue the same query for published records
+        publishedPostReply = await dwn.processMessage(alice.did, publishedPostQuery.message);
+        expect(publishedPostReply.status.code).to.equal(200);
+        expect(publishedPostReply.entries?.length).to.equal(2);
+        const returnedRecordIds = publishedPostReply.entries?.map(e => e.recordId);
+
+        // ensure that both records now exist in results
+        expect(returnedRecordIds).to.have.members([ publishedWrite.message.recordId, draftWrite.message.recordId ]);
+
+        // query after publishing from non owner
+        notOwnerPublishedPostReply = await dwn.processMessage(alice.did, anonymousPostQuery.message);
+        expect(notOwnerPublishedPostReply.status.code).to.equal(200);
+        expect(notOwnerPublishedPostReply.entries?.length).to.equal(2);
+        const nonOwnerReturnedRecordIds = notOwnerPublishedPostReply.entries?.map(e => e.recordId);
+        expect(nonOwnerReturnedRecordIds).to.have.members([ publishedWrite.message.recordId, draftWrite.message.recordId ]);
+
+        // anonymous query after publishing
+        anonymousPublishedPostReply = await dwn.processMessage(alice.did, anonymousPostQuery.message);
+        expect(anonymousPublishedPostReply.status.code).to.equal(200);
+        expect(anonymousPublishedPostReply.entries?.length).to.equal(2);
+        const anonymousReturnedRecordIds = anonymousPublishedPostReply.entries?.map(e => e.recordId);
+        expect(anonymousReturnedRecordIds).to.have.members([ publishedWrite.message.recordId, draftWrite.message.recordId ]);
+      });
+
+      it('should be able to query for unpublished records', async () => {
```

Comment:
```suggestion
      it('should be able to query for unpublished records only if authorized', async () => {
```
The rename implies that we test for both positive and negative cases. Alternatively you can take out the test case for Bob as its own. Bottomline is that IMO testing for unauthorized `published: false` is important, so it needs to be signaled somehow that it is being tested in the sea of tests we've got.

Reply:
Hrm, yeah look at it again I think that pulling Bob's case into it's own would describe the specific intent better. I'll do that.